packages/create-llama/templates/types/streaming/express/src/controllers/chat.controller.ts

```diff
@@ -47,14 +47,29 @@ export const chat = async (req: Request, res: Response) => {
 
     const response = await chatEngine.chat(
       lastMessageContent as MessageContent,
-      messages,
+      messages as ChatMessage[],
       true,
     );
 
     // Transform the response into a readable stream
-    const stream = LlamaIndexStream(response);
+    const { stream, data: streamData } = LlamaIndexStream(response, {
+      parserOptions: {
+        image_url: data?.imageUrl,
+      },
+    });
 
-    streamToResponse(stream, res);
+    // Pipe LlamaIndexStream to response
+    const processedStream = stream.pipeThrough(streamData.stream);
+    return streamToResponse(processedStream, res, {
+      headers: {
+        // MUST have the `X-Experimental-Stream-Data: 'true'` header
+        // in response so the client uses the correct parsing logic
+        // https://sdk.vercel.ai/docs/api-reference/stream-data#on-the-server
```

Comment:
```suggestion
        // resonse MUST have the `X-Experimental-Stream-Data: 'true'` header
        // so that the client uses the correct parsing logic, see
        // https://sdk.vercel.ai/docs/api-reference/stream-data#on-the-server
```